app/javascript/hooks/use-latest-data.ts

```diff
@@ -0,0 +1,11 @@
+import { useEffect, useState } from 'react'
+
+export function useLatestData<T>(data: T): T | undefined {
```

Comment:
I'm not entirely sure what this hook is used for. Could you explain?

Reply:
in v2 there used to be a latestData returned by useQuery, since v3 it is not returned, and I've added this to not break things, but now it is safe to remove, so I just did.